tests/subset_ROI_test.py

```diff
@@ -0,0 +1,158 @@
+    hue_end = hsv_color[0] + hue_range / 2
+    hues = np.linspace(hue_start, hue_end, number) % 1.0
+    shifted_HSVs = [np.concatenate(([hue], hsv_color[1:]), axis=0) for hue in hues]
+    rgb_values = [
+        matplotlib.colors.hsv_to_rgb(shifted_hue) for shifted_hue in shifted_HSVs
+    ]
+    rgb_values = np.vstack(rgb_values)
+
+    rgb_values = rgb_values / 255.0
+
+    return rgb_values
+
+
+colors_per_face = np.full((mesh.n_cells, 3), fill_value=0.5)
+IDs_per_face = np.full((mesh.n_cells, 1), fill_value=np.nan)
+
+for i, (name, group) in enumerate(labels_gdf.groupby("name")):
+    num = len(group)
+    gradient = make_color_gradient(
+        REALISTIC_COLOR_DICT[name], num, hue_range=HUE_RANGE_DICT[name]
+    )
+    # Indices into the original dataset
+    IDs = group.index.to_numpy()
+    # TODO rename
+    for ID, color in zip(IDs, gradient):
+        matching = mesh["ID"] == ID
+        colors_per_face[matching, :] = color
+        IDs_per_face[matching, :] = i
+
+t_vecs = (
+    (0, 0, CAM_HEIGHT),
+    (0, CAM_DIST_FROM_CENTER, CAM_HEIGHT),
+    (CAM_DIST_FROM_CENTER, 0, CAM_HEIGHT),
+    (-CAM_DIST_FROM_CENTER, 0, CAM_HEIGHT),
+    (0, -CAM_DIST_FROM_CENTER, CAM_HEIGHT),
+)
+# Create rotations in roll, pitch, yaw convention
+r_vecs = (
+    (180, 180, 0),  # nadir
+    (180, CAM_PITCH, 0),  # oblique
+    (90, CAM_PITCH, 0),  # oblique
+    (270, CAM_PITCH, 0),  # oblique
+    (0, CAM_PITCH, 0),  # oblique
+)
+
+# Create 4x4 transforms
+cam_to_world_transforms = []
+for r_vec, t_vec in zip(r_vecs, t_vecs):
+    r_mat = Rotation.from_euler("ZXY", r_vec, degrees=True).as_matrix()
+    transform = np.eye(4)
+    transform[:3, :3] = r_mat
+    transform[:3, 3] = t_vec
+    cam_to_world_transforms.append(transform)
+
+IMAGE_FOLDER.mkdir(exist_ok=True, parents=True)
+# Note that these files do not exist yet, but will be later created
+image_filenames = [Path(IMAGE_FOLDER, f"img_{i:03d}.png") for i in range(5)]
+
+camera_set = PhotogrammetryCameraSet(
+    cam_to_world_transforms=cam_to_world_transforms,
+    intrinsic_params_per_sensor_type=CAM_INTRINSICS,
+    image_folder=IMAGE_FOLDER,
+    image_filenames=image_filenames,
+)
+
+#create different ROIs to test the camera_set.get_subset_ROI() function
+polygon1 = Polygon([(0, 0), (1, 0), (1, 1), (0, 1)])
+polygon2 = Polygon([(1, 1), (2, 1), (2, 2), (1, 2)])
+multi_polygon = MultiPolygon([polygon1, polygon2])
+polygon_gdf1 = gpd.GeoDataFrame(data = {'name': ['Polygon 1'], 'geometry': [polygon1]})
+polygon_gdf2 =  gpd.GeoDataFrame(data = {'name': ['Polygon 1', 'Polygon 2'], 'geometry': [polygon1, polygon2]})
+multi_polygon_gdf = gpd.GeoDataFrame(data = {'name': ['MultiPolygon 1'], 'geometry': [multi_polygon]})
+# test is_geospatial flag determination
+assert camera_set.get_subset_ROI(ROI=polygon1).get_camera_locations() == camera_set.get_subset_ROI(ROI=polygon1, is_geospatial=False).get_camera_locations(), "geospatial determination is wrong"
```

Comment:
Asserts are a good start, but I'd like to explore `pytest` or another real testing framework. I'm fine if you want to merge this as-is, but can you leave a comment to look into that in the future and also open an issue about it?

Reply:
Issue #92 already talks about testing. Would you like me to update that one or create a new issue?

Reply:
Updating that one is good.